indico/modules/rb_new/controllers/backend/bookings.py

```diff
@@ -63,11 +63,20 @@ def _serialize_booking_details(booking):
     attributes = reservation_details_schema.dump(booking).data
     date_range, occurrences = get_booking_occurrences(booking)
     date_range = [dt.isoformat() for dt in date_range]
-    occurrences = {dt.isoformat(): reservation_details_occurrences_schema.dump(data).data
-                   for dt, data in occurrences.iteritems()}
     booking_details = dict(attributes)
-    booking_details['occurrences'] = occurrences
+    occurrences_dict = dict(bookings={}, cancellations={}, rejections={}, other_bookings={})
+    booking_details['occurrences'] = occurrences_dict
     booking_details['date_range'] = date_range
+    for dt, data in occurrences.iteritems():
+        occ = reservation_details_occurrences_schema.dump(data).data
+        if data[0].is_cancelled:
```

Comment:
It looks like there can be only one occurrence on each date since it's for a single booking. Any idea why we return a list for each day nonetheless? pinging @meluru since you added `get_booking_occurrences`

@mvidalgarcia since it's always exactly one occurrences you can loop like this to use `occ` instead of `data[0]`:

```python
for dt, [occ] in occurrences.iteritems():
```